shared_model/validators/field_validator.cpp

```diff
@@ -212,6 +212,21 @@ namespace shared_model {
       }
     }
 
+    void FieldValidator::validateOldAccountDetailValue(
+        ReasonsGroupType &reason,
+        const boost::optional<interface::types::AccountDetailValueType>
+            &old_value) const {
+      if (old_value) {
+        if (old_value->size() > value_size) {
```

Comment:
should be possible to reuse the detail value validator
```suggestion
       validateAccountDetailValue(reason, old_value.value());
```